kmeta/names.go

```diff
@@ -55,12 +55,20 @@ func ChildName(parent, suffix string) string {
 			if d := longest - len(ret); d > 0 {
 				ret += suffix[:d]
 			}
-			// If due to trimming above we're terminating the string with a `-`,
-			// remove it.
-			return strings.TrimRight(ret, "-")
+			return makeValidName(ret)
 		}
 		//nolint:gosec // No strong cryptography needed.
 		n = fmt.Sprintf("%s%x", parent[:head-len(suffix)], md5.Sum([]byte(parent)))
 	}
 	return n + suffix
 }
+
+// If due to trimming above we're terminating the string with a non-alphanumeric
+// character, remove it.
+func makeValidName(n string) string {
+	match := regexp.MustCompile(`^[a-zA-Z0-9]*$`)
```

Comment:
Also you probably need regexp compiled once, no need to do it per name?

Reply:
ah good point, I can pull it out